test/index.ts

```diff
@@ -0,0 +1,15 @@
+import fs from "node:fs/promises";
+
+async function* findTestFiles(url) {
+  for await (const dirent of await fs.opendir(url)) {
+    if (dirent.name === "node_modules" || dirent.name.startsWith(".")) continue;
+
+    if (dirent.isDirectory())
+      yield* findTestFiles(new URL(`${dirent.name}/`, url));
+    else if (dirent.name.endsWith(".spec.ts")) yield new URL(dirent.name, url);
+  }
+}
+
+for await (const file of findTestFiles(new URL("../", import.meta.url))) {
+  await import(file);
+}
```

Comment:
Please make sure it respect our linter, see mine here: https://github.com/NodeSecure/js-x-ray/pull/108/files#diff-5bb8db779819ddef5956a5d9d5949c05ef7445237656ca37bf2f02720271440b